tests/function/gtCxxProperties.cpp

```diff
@@ -144,3 +144,16 @@ TEST(LibopaecppPropsCommonALL, get_num_errors) {
   p->num_errors = 9;
   EXPECT_EQ(static_cast<uint32_t>(p->num_errors), 9);
 }
+
+/**
+ * @test segment
+ * Given a properties properties object with the segment property set to a
+ * known value
+ * When I get the segment property
+ * Then the number is the expected value
+ */
+TEST(LibopaecppPropsCommonALL, get_segment) {
+  auto p = properties::get();
+  p->segment = 9090;
+  EXPECT_EQ(static_cast<uint16_t>(p->segment), 9090);
```

Comment:
So, you're assigning a value to a variable, then expecting it to be equal to the value you just assigned to it?  I don't see where this is actually testing anything, but there may be some C++ magic here somewhere.  Can you clarify?

Reply:
line 157 is exercising the assignment operator in the pvalue template class which will ultimately call fpgaPropertiesSetSegment.
line 158 is exercising the conversion operator in the pvalue template classs which will call fpgaPropertiesGetSegment to retrieve the value.